lib/scraped_page/request.rb

```diff
@@ -0,0 +1,26 @@
+require 'scraped_page/strategy/live_request'
+require 'scraped_page/response'
+
+class ScrapedPage
+  class Request
+    def initialize(url:, readers: [Strategy::LiveRequest.new])
+      @url = url
+      @readers = readers
+    end
+
+    def response(processors = [])
+      abort "Failed to fetch #{url}" if first_successful_response.nil?
```

Comment:
Tiny nit, but `first_successful_response.nil?` caused me a double-take, as it read more like "We got a response, but it was nil" rather than "we didn't get any response". Perhaps `unless first_successful_response` would be clearer?